iceoryx2-cli/tests/config_description_tests.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2025 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+mod config_descriptions_tests {
+    use std::collections::HashSet;
+
+    use ron::de::from_str;
+    use ron::ser::to_string;
+    use ron::Value;
+
+    use iceoryx2::config::Config;
+    use iceoryx2_bb_testing::assert_that;
+    use iceoryx2_cli::config_descriptions::get_sections;
+
+    // Recursively walk through ron::Value to flatten all keys into a HashSet
+    fn collect_keys_ron(value: &Value, prefix: String, keys: &mut HashSet<String>) {
+        match value {
+            Value::Map(map) => {
+                for (k, v) in map.iter() {
+                    let k_str = match k {
+                        Value::String(s) => s.clone(),
+                        _ => continue, // skip non-string keys
+                    };
+
+                    let new_prefix = if prefix.is_empty() {
+                        k_str
+                    } else {
+                        format!("{}.{}", prefix, k_str)
+                    };
+
+                    collect_keys_ron(v, new_prefix, keys);
+                }
+            }
+            _ => {
+                keys.insert(prefix);
+            }
+        }
+    }
+
+    #[test]
+    fn check_config_description_is_present_for_all_config_parameters() {
+        // Replace this with a version of Config that populates all Option<T> fields
+        let config = Config::default();
```

Comment:
Does this mean that `RON` also does not show `Option<T>` with a value of `None`?